### ISSUE TITLE
Agent unable to connect to local collector

### ISSUE THREAD

**Issue by mnaga** [jaegertracing/jaeger#261]
I have a collector running locally. And am starting a agent with command line options for my collector server/port as below. Couple of questions

1) Am not sure why the service discovery is enabled when I am explicitly passing the collector info
2) How do I met more info on the tchannel error?

$ go run ./cmd/agent/main.go -collector.host-port 127.0.0.1:14267
{"level":"info","ts":1499793930.997451,"caller":"tchannel/bulider.go:95","msg":"Enabling service discovery","service":"jaeger-collector"}
{"level":"info","ts":1499793930.9975188,"caller":"peerlistmgr/peer_list_mgr.go:117","msg":"Registering active peer","peer":"127.0.0.1:14267"}
{"level":"info","ts":1499793930.9978952,"caller":"agent/main.go:52","msg":"Starting agent"}
{"level":"info","ts":1499793931.998254,"caller":"peerlistmgr/peer_list_mgr.go:165","msg":"Not enough connected peers","connected":0,"required":1}
{"level":"info","ts":1499793931.998381,"caller":"peerlistmgr/peer_list_mgr.go:172","msg":"Trying to connect to peer","host:port":"127.0.0.1:14267"}

**Comment by yurishkuro** [comment]
Can you try running all-in-one from source? The components still communicate over the same ports.

```
$ go run ./cmd/standalone/main.go --span-storage.type=memory
{"level":"info","ts":1499795331.6866407,"caller":"tchannel/bulider.go:95","msg":"Enabling service discovery","service":"jaeger-collector"}
{"level":"info","ts":1499795331.6867087,"caller":"peerlistmgr/peer_list_mgr.go:117","msg":"Registering active peer","peer":"127.0.0.1:14267"}
{"level":"info","ts":1499795331.687073,"caller":"standalone/main.go:82","msg":"Starting agent"}
{"level":"info","ts":1499795331.68803,"caller":"standalone/main.go:117","msg":"Starting jaeger-collector TChannel server","port":14267}
{"level":"info","ts":1499795331.6881118,"caller":"standalone/main.go:124","msg":"Starting jaeger-collector HTTP server","http-port":14268}
{"level":"info","ts":1499795331.69117,"caller":"standalone/main.go:174","msg":"Starting jaeger-query HTTP server","port":16686}
{"level":"info","ts":1499795332.689505,"caller":"peerlistmgr/peer_list_mgr.go:165","msg":"Not enough connected peers","connected":0,"required":1}
{"level":"info","ts":1499795332.68965,"caller":"peerlistmgr/peer_list_mgr.go:172","msg":"Trying to connect to peer","host:port":"127.0.0.1:14267"}
{"level":"info","ts":1499795332.6903925,"caller":"peerlistmgr/peer_list_mgr.go:177","msg":"Connected to peer","host:port":"[::]:14267"}
```

**Comment by jnewmano** [comment]
There is an error generated (https://github.com/uber/jaeger/blob/master/pkg/discovery/peerlistmgr/peer_list_mgr.go#L174) that is not being logged anywhere. 

Logging this error message would be beneficial to diagnosing connection issues. We are currently running into some sort of connection issue where the connection is taking longer than connCheckTimeout (250ms). 

Would a PR with our changes to this part of code be beneficial, if so is there a place to look at for PR guidelines?

**Comment by yurishkuro** [comment]
@jnewmano a PR would be great - see https://github.com/uber/jaeger/blob/master/CONTRIBUTING.md

**Comment by yurishkuro** [comment]
I think this has been resolved by the attached PR #356, closing.
